examples/log2har.rs

```diff
@@ -0,0 +1,331 @@
+{
+    pretty_env_logger::init();
+    let status = match main_() {
+        Ok(_) => 0,
+        Err(e) => {
+            eprintln!("Error: {}", e);
+            for c in e.iter_chain().skip(1) {
+                eprintln!(" Caused by {}", c);
+            }
+            eprintln!("{}", e.backtrace());
+            1
+        }
+    };
+    ::std::process::exit(status);
+}
+
+/// Collects all the given inputs as kubectl -v8 web transactions and outputs
+/// as one har.
+/// `kubectl version -v8 2> version.log && log2har version.log > version.har`
+fn main() {
+    main_result(main_)
+}
+
+#[cfg(test)]
+mod test {
+    #[test]
+    fn parse_v8() {
+        let input = r#"I0225 20:16:03.123108   14148 round_trippers.go:383] GET https://localhost:6445/version?timeout=32s
+I0225 20:16:03.123108   14148 round_trippers.go:390] Request Headers:
+I0225 20:16:03.123108   14148 round_trippers.go:393]     Accept: application/json, */*
+I0225 20:16:03.123108   14148 round_trippers.go:393]     User-Agent: kubectl/v1.13.0 (windows/amd64) kubernetes/ddf47ac
+I0225 20:16:03.146122   14148 round_trippers.go:408] Response Status: 200 OK in 23 milliseconds
+I0225 20:16:03.146122   14148 round_trippers.go:411] Response Headers:
+I0225 20:16:03.146122   14148 round_trippers.go:414]     Content-Type: application/json
+I0225 20:16:03.146122   14148 round_trippers.go:414]     Content-Length: 263
+I0225 20:16:03.146122   14148 round_trippers.go:414]     Date: Thu, 21 Feb 2019 21:04:01 GMT
+I0225 20:16:03.147299   14148 request.go:942] Response Body: {
+  "major": "1",
+  "minor": "13",
+  "gitVersion": "v1.13.0",
+  "gitCommit": "ddf47ac13c1a9483ea035a79cd7c10005ff21a6d",
+  "gitTreeState": "clean",
+  "buildDate": "2018-12-03T20:56:12Z",
+  "goVersion": "go1.11.2",
+  "compiler": "gc",
+  "platform": "linux/amd64"
+}
+"#;
+        let expected = r#"{"pageref":null,"startedDateTime":"","time":0,"request":{"method":"GET","url":"https://localhost:6445/version?timeout=32s","httpVersion":"unknown","cookies":[],"headers":[{"name":"Accept","value":"application/json, */*"},{"name":"User-Agent","value":"kubectl/v1.13.0 (windows/amd64) kubernetes/ddf47ac"}],"queryString":[],"headersSize":-1,"bodySize":-1},"response":{"status":-1,"statusText":"","httpVersion":"unknown","cookies":[],"headers":[{"name":"Content-Type","value":"application/json"},{"name":"Content-Length","value":"263"},{"name":"Date","value":"Thu, 21 Feb 2019 21:04:01 GMT"}],"content":{"size":-1,"mimeType":"","text":"ewogICJtYWpvciI6ICIxIiwKICAibWlub3IiOiAiMTMiLAogICJnaXRWZXJzaW9uIjogInYxLjEzLjAiLAogICJnaXRDb21taXQiOiAiZGRmNDdhYzEzYzFhOTQ4M2VhMDM1YTc5Y2Q3YzEwMDA1ZmYyMWE2ZCIsCiAgImdpdFRyZWVTdGF0ZSI6ICJjbGVhbiIsCiAgImJ1aWxkRGF0ZSI6ICIyMDE4LTEyLTAzVDIwOjU2OjEyWiIsCiAgImdvVmVyc2lvbiI6ICJnbzEuMTEuMiIsCiAgImNvbXBpbGVyIjogImdjIiwKICAicGxhdGZvcm0iOiAibGludXgvYW1kNjQiCn0K","encoding":"base64"},"redirectURL":"","headersSize":-1,"bodySize":-1},"cache":{},"timings":{"send":-1,"wait":-1,"receive":-1}}
```

Comment:
This long line is unreadable (for me, particularly through the github code browser).  Can we put this in some sort of structured/indented object instead (even if just `json!()`)?
Related: should `parse_log` return a typed object rather than a string?

Reply:
I can parse both into a HAR or  json!().

parse_+log is returning a string just for convenience. An iterator over parsed items - plain entries + parsed requests-as-hars - would make a great deal of sense as a refactor, but not really reusable at this point; if we move it into a package somewhere I think it would be worth doing that.